plugins/inputs/github/github.go

```diff
@@ -71,9 +77,16 @@ func (g *GitHub) createGitHubClient(ctx context.Context) (*github.Client, error)
 
 		g.obfusticatedToken = g.AccessToken[0:4] + "..." + g.AccessToken[len(g.AccessToken)-3:]
 
-		return github.NewClient(oauthClient), nil
+		return g.newGithubClient(oauthClient)
 	}
 
+	return g.newGithubClient(httpClient)
+}
+
+func (g *GitHub) newGithubClient(httpClient *http.Client) (*github.Client, error) {
+	if g.EnterpriseBaseURL != "" && g.EnterpriseUploadURL != "" {
```

Comment:
```suggestion
	if g.EnterpriseBaseURL != "" {
```
I think it's actually ok for me to allow EnterpriseUploadURL to be `""`. It works fine (tested it and by looking at the code). I don't think the UploadURL will be necessary for these stats anyway. It's basically only used for `NewUploadRequest` as far as I can tell.